jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/DisableBasicAuthenticationRule.java

```diff
@@ -0,0 +1,154 @@
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.RESOURCE_ID), resourceId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+            try {
+                isKeyEnabled = checkIfBasicAuthIsDisabled(vmEsURL, mustFilter);
+                if (!isKeyEnabled) {
+                    List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                    LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, description);
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, violationReason);
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========rule ended with status failure {}", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                            annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("======== ended with status true=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+
+    private boolean checkIfBasicAuthIsDisabled(String vmEsURL, Map<String, Object> mustFilter) throws Exception {
+        logger.debug("========checkIfBasicAuthIsDisabled  started=========");
+        JsonArray hitsJsonArray = GCPUtils.getHitsArrayFromEs(vmEsURL, mustFilter);
+        boolean validationResult = true;
+        if (hitsJsonArray.size() > 0) {
+            logger.debug("========checkIfBasicAuthIsDisabled hit array=========");
+
+            JsonObject gkeCluster = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                    .get(PacmanRuleConstants.SOURCE);
+
+            logger.debug("Validating the data item: {}", gkeCluster);
+
+            String username = gkeCluster.getAsJsonPrimitive("username").getAsString();
+            String password = gkeCluster.getAsJsonPrimitive("password").getAsString();
+            String version=gkeCluster.getAsJsonPrimitive("version").getAsString().substring(0,4);
+            if(versionCompare(version,"1.19")<0 &&!username.isEmpty() && !password.isEmpty())
+            {
+                validationResult=false;
+            }
+
+        }
+
+        return validationResult;
+    }
+    // Method to compare two versions. Returns 1 if v2 is smaller, -1 if v1 is smaller, 0 if equal
+    static int versionCompare(String v1, String v2)
```

Comment:
To compare the version, we can parse the strings to Float and use Float.compare

Reply:
fixed logic for version comparison